cirq-core/cirq/ops/projector_test.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import pytest
+
+import cirq
+
+
+def test_projector_qid():
+    q0 = cirq.NamedQubit('q0')
+
+    zero_projector = cirq.ProjectorString({q0: 0})
+    one_projector = cirq.ProjectorString({q0: 1})
+
+    np.testing.assert_allclose(zero_projector.matrix().toarray(), [[1.0, 0.0], [0.0, 0.0]])
+    np.testing.assert_allclose(one_projector.matrix().toarray(), [[0.0, 0.0], [0.0, 1.0]])
+
+
+def test_projector_repr():
+    q0 = cirq.NamedQubit('q0')
+
+    assert (
+        repr(cirq.ProjectorString({q0: 0}))
+        == "cirq.ProjectorString(projector_dict={cirq.NamedQubit('q0'): 0},coefficient=(1+0j))"
+    )
+
+
+def test_projector_from_np_array():
+    q0 = cirq.NamedQubit('q0')
+
+    zero_projector = cirq.ProjectorString({q0: 0})
+    np.testing.assert_allclose(zero_projector.matrix().toarray(), [[1.0, 0.0], [0.0, 0.0]])
+
+
+def test_projector_matrix_missing_qid():
```

Comment:
Probably want another matrix test that verifies some slightly more nontrivial cases. Certainly ones that include a coefficient value other than 1.

Reply:
Added the test in `test_projector_qid` above. I also took the liberty of renaming it `test_projector_matrix` since it wasn't a good name.

Reply:
Added a test in `test_projector_matrix_missing_qid` and in `test_projector_matrix`